### DISCUSSION TITLE
Add BaseHTTPMiddleware to FastAPI

### DISCUSSION THREAD

**Discussion by aryaniyaps** [fastapi/fastapi#8541] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from starlette.middleware.base import BaseHTTPMiddleware

class MyMiddleware(BaseHTTPMiddleware):
    async def dispatch(self, request, call_next):
        pass


def create_app()
    app = FastAPI()
    app.add_middleware(MyMiddleware)
    return app

app = create_app()
```

### Description

I usually split my fastapi code across multiple files. I usually have an application factory function, which incrementally adds routes, event handlers and middleware to the app.

This means that the middleware lies in another file, and the app factory lies in another file.

Therefore, it's not possible to define middleware using decorators, like `@app.middleware`.

In order to use `app.add_middleware`, my middleware must inherit BaseHTTPMiddleware. However, this is not present in FastAPI right now.

### Wanted Solution

I would like to add BaseHTTPMiddleware to FastAPI (Basically import it from starlette in the file `fastapi.middleware.base`), for the sake of developer convonience.

With that in place, we can import BaseHTTPMiddleware like:

### Wanted Code

```python
from fastapi import FastAPI
from fastapi.middleware.base import BaseHTTPMiddleware

class MyMiddleware(BaseHTTPMiddleware):
    async def dispatch(self, request, call_next):
        pass


def create_app()
    app = FastAPI()
    app.add_middleware(MyMiddleware)
    return app

app = create_app()
```

### Operating System

Windows

### FastAPI Version

0.71.0

### Python Version

3.9

**Comment by TBBle** [comment, marked as answer]
Is there a particular reason you don't just import it directly from Starlette, since FastAPI depends on Starlette?
